tests/src/test/scala/ch/renku/acceptancetests/workflows/Environments.scala

```diff
@@ -0,0 +1,125 @@
+package ch.renku.acceptancetests.workflows
+
+import ch.renku.acceptancetests.model.projects.{ProjectDetails, ProjectIdentifier}
+import ch.renku.acceptancetests.pages._
+import ch.renku.acceptancetests.tooling.{AcceptanceSpec, AnonEnvConfig, DocsScreenshots}
+
+import scala.concurrent.duration._
+import scala.language.postfixOps
+
+trait Environments {
+  self: AcceptanceSpec =>
+
+  def launchEnvironment(implicit projectDetails: ProjectDetails, docsScreenshots: DocsScreenshots): JupyterLabPage = {
+    implicit val projectPage = ProjectPage()
+    When("user clicks on the Environments tab")
+    click on projectPage.Environments.tab
+    docsScreenshots.reachedCheckpoint()
+
+    clickNewAndWaitForImageBuild
+    docsScreenshots.reachedCheckpoint()
+
+    clickStartEnvironmentAndWaitForReady
+    docsScreenshots.reachedCheckpoint()
+
+    projectPage.Environments.Running.connectToJupyterLab
+
+    Then("a JupyterLab page is opened on a new tab")
+    val jupyterLabPage = JupyterLabPage()
+    verify browserSwitchedTo jupyterLabPage sleep (5 seconds)
+    return jupyterLabPage
+  }
+
+  def stopEnvironment(implicit projectDetails: ProjectDetails): Unit =
+    stopEnvironmentOnProjectPage(ProjectPage())
+
+  def stopEnvironment(projectId: ProjectIdentifier): Unit =
+    stopEnvironmentOnProjectPage(ProjectPage(projectId))
+
+  def stopEnvironmentOnProjectPage(projectPage: ProjectPage): Unit = {
+    When("the user switches back to the Renku tab")
+    verify browserSwitchedTo projectPage
+    And("the Environments tab")
+    click on projectPage.Environments.tab
+    And("they turn off the interactive session they started before")
+    click on projectPage.Environments.Running.connectDotButton
+    click on projectPage.Environments.Running.stopButton sleep (10 seconds)
+    Then("the session gets stopped and they can see the New Session link")
+    verify userCanSee projectPage.Environments.newLink
+  }
+
+  def launchAnonymousEnvironment(anonEnvConfig: AnonEnvConfig): Option[JupyterLabPage] = {
```

Comment:
What do you think about making the `anonEnvConfig: AnonEnvConfig` `implicit`? It could make the spec a bit cleaner. Alternatively, as the `AnonEnv` is, I guess build for the `Environments`, we could potentially take it from the `MinimalFunctionalitySpec` and mix it in directly to the Environments. Perhaps, that could be even better than the implicit on the config. That would allow us to take the config args from all the `Environment` defs.

Reply:
I have made this change.

I didn't initially implement it this way because I thought I would use different projects for the unprivileged test and anonymous test, but they now use the same project, so there is no reason against making the Config implicit anymore.